### ISSUE TITLE
[file-search] editor is added to 'recently opened' only once it has focus / activated

### ISSUE THREAD

**Issue by vince-fugnitto** [theia-ide/theia#4388]
There is a discrepancy with our behaviour and that of `vscode` when registering an 
editor as **recently opened**. 

In `vscode` as soon as an editor is opened (preview or not) it is immediately listed as **recently opened**.
In `theia` it takes much more effort and isn't as reliable.

**Comment by kittaakos** [comment]
@vince-fugnitto, nice, I was unable to reproduce the defect consistently, I am glad you experienced the same oddity.

----

I want to hijack this thread for a moment and add a tiny remark: I do not like that the "diff-editors" show up among the recently opened items just because we register an editor history entry for them. 

![screen shot 2019-02-22 at 13 47 15](https://user-images.githubusercontent.com/1405703/53243521-6a757900-36a8-11e9-8476-c11da2f33846.jpg)

Does it worth a separate issue? Can we change it or it's only me who finds this behavior strange. As a workaround, I can use [`Clear Editor History`](https://github.com/theia-ide/theia/pull/4392) to get rid of the diff editors from time to time.

**Comment by vince-fugnitto** [comment]
@kittaakos I also noticed the issue regarding the `diff-editors` registering in our `recently opened`. 
Personally I don't like it being there, also vscode does not display it so it's even more reason to 
fix the issue :)
